caffe/src/caffe/layers/molgrid_data_layer.cpp

```diff
@@ -1077,6 +1077,44 @@ void MolGridDataLayer<Dtype>::setReceptor(const vector<atom>& receptor, const ve
   batch_info[0].setReceptor(rec);
 }
 
+template <typename Dtype>
+void MolGridDataLayer<Dtype>::setReceptorFlex(const vector<atom>& receptor, const vector<vec>& flexcoords) {
+  CHECK_GT(batch_info.size(), 0) << "Empty batch info in setReceptor";
+
+  vector<float3> cs; cs.reserve(receptor.size());
+  vector<float> types; types.reserve(receptor.size());
+  vector<float> radii; radii.reserve(receptor.size());
+
+  //receptor atoms
+  for (unsigned i = 0, n = receptor.size(); i < n; i++) {
+    const atom& a = receptor[i];
+    smt origt = a.sm;
+    auto t_r = recTypes->get_int_type(origt);
+    int t = t_r.first;
+
+    vec coord;
+    if(i < flexcoords.size()){
+      // Get flex and inflex coords explicitly
+      coord = flexcoords[i];
+    }else{
+      // Get standard receptor coordinate from atom
+      coord = a.coords;
+    }
+
+    cs.push_back(gfloat3(coord[0],coord[1],coord[2]));
+    types.push_back(t);
+    radii.push_back(t_r.second);
+
+    if(t < 0 && origt > 1) { //don't warn about hydrogens
+      std::cerr << "Unsupported receptor atom type " << GninaIndexTyper::gnina_type_name(origt) << "\n";
+    }
+  }
+
+  CoordinateSet rec(cs, types, radii, recTypes->num_types());
+
```

Comment:
Removed rotations, since the receptor is automatically frozen.

Reply:
why is this necessary? if flexible residues aren't being rotated correctly, that should be fixed - and the fix isn't to turn off rotations. I'm wondering if the rigid body rotations looked like they weren't working because you were using the coordinates from the `atoms` vector instead of the ones in `coords` - not sure why they wouldn't just work, but if they aren't, that should be fixed; don't like this freezing business.

Reply:
As you suggest, the problem with rotations was probably caused by the wrong coordinates (to be checked). However, I'm probably missing something but I don't see how rotations would come into place when locally optimizing the system; do you want to do multiple minimizations with different orientations?

Reply:
Actually, the problem with rotations persists even with the correct coordinates. I'll try to figure out why and how to fix it.